File: Challenges/5kyu/rgb_to_hex_conversion/php/solution.php
function rgb($r, $g, $b) {
  $hex = "";
  
  $hex .= byteToHex($r);
  $hex .= byteToHex($g);
  $hex .= byteToHex($b);
  
  return $hex;
}

function byteToHex($byte) {
  $digits = "0123456789ABCDEF";
  $hex = "";
  
  $byte = ($byte < 0) ? 0 : (($byte > 255) ? $byte = 255 : $byte);
  $hex .= $digits[($byte - ($byte % 16)) / 16];
  $hex .= $digits[$byte % 16];
  
  return $hex;
}

File: Challenges/5kyu/rgb_to_hex_conversion/java/solution.java
public class RgbToHex {

  public static String rgb(int r, int g, int b) {
    String hex = "";

    hex += RgbToHex.byteToHex(r);
    hex += RgbToHex.byteToHex(g);
    hex += RgbToHex.byteToHex(b);

    return hex;
  }

  public static String byteToHex(int intByte) {
    String digits = "0123456789ABCDEF";
    String hex = "";

    intByte = (intByte < 0) ? 0 : (intByte > 255) ? intByte = 255 : intByte;
    hex += digits.charAt((intByte - (intByte % 16)) / 16);
    hex += digits.charAt(intByte % 16);

    return hex;
  }
}